packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1035,8 +1035,12 @@ export class TransactionController extends BaseController<
       updateTransactionHistory(transactionMeta, note);
     }
     const index = transactions.findIndex(({ id }) => transactionMeta.id === id);
+    const prevTransactionStatus = transactions[index]?.status;
     transactions[index] = transactionMeta;
     this.update({ transactions: this.trimTransactionsForState(transactions) });
+    if (prevTransactionStatus !== transactionMeta.status) {
+      this.hub.emit('transaction-status-update', { transactionMeta });
```

Comment:
Just for audit sake, the extension arguments were the ID and the status but I really like this since it gives the listener both of those plus anything else they may want to know.

Reply:
Exactly, we kinda have a standart `transactionMeta` included in the events right now, I didn't want to break it tbh